### ISSUE TITLE
Game crashes with a "metric not found" error on song fail

### ISSUE THREAD

**Issue by ghost** [dguzek/Simply-Love-SM5#110]
Specifically, it gives this error: `Metric "PlayerStageStats::GradePercentFailed" could not be found in "Themes/Simply Love/metrics.ini"' or "Themes/_fallback/metrics.ini".`

I was playing on SM 5.0.12, so I assume that could be the reason.

**Comment by ElliottBelardo** [comment]
- Event mode was on.
- Coin mode was set to Home.
- Fail type was set to Immediate, but I was able to reproduce the crash with fail type set to ImmediateContinue.

I assume that's all the important information needed.

**Comment by dguzek** [comment]
In that no one has ever reported this problem and I am unable to reproduce, it may take some time to figure what is different about your configuration that is causing StepMania to crash.  Can you attach or upload a copy of the crashlog.txt that is generated by StepMania?

**Comment by ElliottBelardo** [comment]
[log.txt](https://github.com/dguzek/Simply-Love-SM5/files/1980028/log.txt)

**Comment by dguzek** [comment]
Thanks.  Unfortunately, I gleaned nothing from that other than the fact that you are using an older version of StepMania (5.0.12 when 5.1.0 is out) that I can test later.

I don't anticipate finding any problem there, however.  What modifiers are you using?  In particular, are you using the target score system?  I didn't write that, don't know what it does or how it works, and there are possibly bugs lurking in it.

**Comment by ElliottBelardo** [comment]
The target score graph was enabled, and the target was set to PB. So I assume the target score was part of the issue, considering I hadn't set a score on the simfile I was playing yet.

**Comment by dguzek** [comment]
@ElliottBelardo – Can you try changing [this line](https://github.com/dguzek/Simply-Love-SM5/blob/00fdf7112b0050d229679c39d777dfca5f0bb11a/BGAnimations/ScreenGameplay%20underlay/PerPlayer/TargetScore/default.lua#L275) so that it is wrapped in a check like this:

```lua
if currentGrade ~= "Grade_Failed" then
	self:queuecommand("GradeChanged")
end
```

Please let me know if that resolves the issue for you.

**Comment by dguzek** [comment]
@ElliottBelardo – Any update on this?

**Comment by ElliottBelardo** [comment]
I'm not entirely sure whether or not that fixed the bug. First, I tried the fix and it worked. Then, I played using the unfixed version, and it also worked (presumably because it already wrote the high score to the game data). So it's entirely possible that the game crash is caused by the game trying to call a nonexistent value (the personal record on the simfile).

**Comment by dguzek** [comment]
Thanks for responding with this new information.  I'll poke at it some more.

**Comment by dguzek** [comment]
This has (hopefully) been resolved for the v4.8 release.  See d0603637099846fde3a0e0baf67e3778c0a5ba30 for more detailed information.
